cmd/oras/internal/display/metadata/text/manifest_index.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package text
+
+import (
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras/cmd/oras/internal/display/metadata"
+	"oras.land/oras/cmd/oras/internal/output"
+	"oras.land/oras/internal/contentutil"
+	"oras.land/oras/internal/descriptor"
+)
+
+// ManifestIndexCreateHandler handles text metadata output for index create events.
+type ManifestIndexCreateHandler struct {
+	printer *output.Printer
+}
+
+// NewManifestIndexCreateHandler returns a new handler for index create events.
+func NewManifestIndexCreateHandler(printer *output.Printer) metadata.ManifestIndexCreateHandler {
+	return &ManifestIndexCreateHandler{
+		printer: printer,
+	}
+}
+
+// OnIndexPacked implements ManifestIndexCreateHandler.
+func (h *ManifestIndexCreateHandler) OnIndexPacked(desc ocispec.Descriptor) error {
+	return h.printer.Println("Packed", descriptor.ShortDigest(desc), ocispec.MediaTypeImageIndex)
+}
+
+// OnIndexPushed implements ManifestIndexCreateHandler.
+func (h *ManifestIndexCreateHandler) OnIndexPushed(path string) error {
+	return h.printer.Println("Pushed", path)
+}
+
+// OnTagged implements TaggedHandler.
+func (h *ManifestIndexCreateHandler) OnTagged(_ ocispec.Descriptor, tag string) error {
+	return h.printer.Println("Tagged", tag)
+}
+
+// OnCompleted implements ManifestIndexCreateHandler.
+func (h *ManifestIndexCreateHandler) OnCompleted(desc ocispec.Descriptor) error {
+	return h.printer.Println("Digest:", desc.Digest)
+}
+
+// ManifestIndexUpdateHandler handles text metadata output for index update events.
+type ManifestIndexUpdateHandler struct {
+	printer *output.Printer
```

Comment:
Can you help separate `ManifestIndexUpdateHandler` and `ManifestIndexCreateHandler` into `manifest_index_update.go` and `manifest_index_create.go`?